h/sentry/helpers/__init__.py

```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
```

Comment:
Is there a standard Python convention for package-private modules? Do we already have a convention in our code? [Google's styleguide](http://google.github.io/styleguide/pyguide.html#3163-file-naming) suggests underscore prefixes.

Reply:
The loose convention I've been following is to look in a package's top-level `__init__.py` for the package's public API. So for `h/sentry/__init__.py` that'd be `report_exception()` and `includeme()`. Other names that're imported into `__init__.py` (`unicode_literals`, `sentry_sdk`) are obviously not meant to be public.

Generally other packages should import names from a top-level package's `__init__.py` and not import submodules directly. When a name from a submodule is intended to be public that gets imported into the API. So for example its `from h.models import User` not `from h.models.user import User`. When an `__init__.py` also has to import some stuff that it doesn't intend to make public, and it's not totally obvious that that isn't the intention, then we use an `__all__` tuple in `__init__.py` to explicitly define the list of public names.

I think the correct thing to do [according to PEP 8](https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces) is to rename the `helpers` directory to `_helpers`:

> Even with `__all__` set appropriately, internal interfaces (packages, modules, classes, functions, attributes or other names) should still be prefixed with a single leading underscore.

Note that while `h.sentry.helpers` is a private package of `h.sentry` (and so it should be `_helpers`), `before_send.py`, `event.py` and `filters.py` are _not_ private packages of `h.sentry.helpers`, they're meant to be imported and used by top-level `h.sentry` code, so those wouldn't need underscores.

We don't follow this pattern of naming private packages and modules with leading underscores in our Python code currently. But perhaps we should do.

I'm not sure that Python code generally follows this or any other pattern for defining package and module APIs in a consistent conventional way. But looking at both Google's guide and PEP 8 it looks like if you wanted to do it, leading underscores would be the way.